pkg/etcd/client.go

```diff
@@ -57,8 +57,10 @@ func NewFromConfig(ctx context.Context, vConfig *config.VirtualClusterConfig) (C
 
 		if vConfig.ControlPlane.BackingStore.Etcd.Embedded.Enabled {
 			etcdEndpoints = "https://127.0.0.1:2379"
-		} else {
+		} else if vConfig.ControlPlane.BackingStore.Etcd.Deploy.Service.Enabled {
 			etcdEndpoints = "https://" + vConfig.Name + "-etcd:2379"
+		} else if vConfig.ControlPlane.BackingStore.Etcd.Deploy.HeadlessService.Enabled {
```

Comment:
Can a scenario exist in which `Deploy.Enable` is true, but neither `Service.Deploy` nor `HeadlessService.Enabled` is true?

Reply:
Yes, It is possible to reach this situation when both the above fields are set to disabled, and subsequently this would lead to a critical failure. 

On a related note, I have another thought. Given that the headless service is the [governing service](https://github.com/loft-sh/vcluster/blob/94e59b63d9e3bc82c0c55b2ffaed35ea4881f9d4/chart/templates/etcd-statefulset.yaml#L26) for the etcd statefulset and is required to be present whenever the statefulset exists, does it then make more sense to remove the `headlessservice.enabled` field and have its creation defined primarily by `Deploy.Enable`. This will prevent it from being manually disabled by someone like above. Also, this will further simplify all the checks I added in this PR, where we will have the headless version as the last/fallback option under the else clause.

Reply:
Changes are done now.